FMS/Pages/Facilities/Index.cshtml.cs

```diff
@@ -1,75 +1,112 @@
-
-        // Shows results section after searching
-        public bool ShowResults { get; private set; }
-
-        // Select Lists
-        public SelectList Counties => new SelectList(Data.Counties, "Id", "Name");
-        public SelectList States => new SelectList(Data.States);
-        public SelectList FacilityStatuses { get; private set; }
-        public SelectList FacilityTypes { get; private set; }
-        public SelectList BudgetCodes { get; private set; }
-        public SelectList OrganizationalUnits { get; private set; }
-        public SelectList ComplianceOfficers { get; private set; }
-
-        public IndexModel(
-            IFacilityRepository repository,
-            ISelectListHelper listHelper)
-        {
-            _repository = repository;
-            _listHelper = listHelper;
-        }
-
-        public async Task<IActionResult> OnGetAsync()
-        {
-            Spec = new FacilitySpec();
-            await PopulateSelectsAsync();
-            return Page();
-        }
-
-        public async Task<IActionResult> OnGetSearchAsync(FacilitySpec spec, [FromQuery] int p = 1)
-        {
-            // Get the list of facilities matching the "Spec" criteria
-            FacilityList = await _repository.GetFacilityPaginatedListAsync(spec, p, GlobalConstants.PageSize);
-            Spec = spec;
-            ShowResults = true;
-            await PopulateSelectsAsync();
-            return Page();
-        }
-
-        public IActionResult OnPost(FacilitySpec spec)
-        {
-            return RedirectToPage("../Reports/Index", spec);
-        }
-
-        private async Task PopulateSelectsAsync()
-        {
-            BudgetCodes = await _listHelper.BudgetCodesSelectListAsync();
-            ComplianceOfficers = await _listHelper.ComplianceOfficersSelectListAsync();
-            FacilityStatuses = await _listHelper.FacilityStatusesSelectListAsync();
-            FacilityTypes = await _listHelper.FacilityTypesSelectListAsync();
-            OrganizationalUnits = await _listHelper.OrganizationalUnitsSelectListAsync();
-        }
-    }
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CsvHelper.Configuration;
+using FMS.Domain.Data;
+using FMS.Domain.Dto;
+using FMS.Domain.Dto.PaginatedList;
+using FMS.Domain.Repositories;
+using FMS.Platform.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace FMS.Pages.Facilities
+{
+    public class IndexModel : PageModel
+    {
+        private readonly IFacilityRepository _repository;
+        private readonly ISelectListHelper _listHelper;
+
+        // "Spec" is the Facility DTO bound to the HTML Page elements
+        public FacilitySpec Spec { get; set; }
+
+        // "ReportSpec" is the Facility DTO Bound to Export Results to Excel (csv file)
+        [BindProperty]
+        public FacilitySpec ReportSpec { get; set; }

```

Comment:
Don't create a new `FacilitySpec` property here. Bind the existing `Spec` instead.